src/algebraic_topology/extra_degeneracy.lean

```diff
@@ -0,0 +1,188 @@
+/-
+Copyright (c) 2022 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+
+import algebraic_topology.simplicial_set
+import tactic.fin_cases
+
+/-!
+
+# Augmented simplicial objects with an extra degeneracy
+
+In simplicial homotopy theory, in order to prove that the connected components
+of a simplicial set `X` are contractible, it suffices to construct an extra
+degeneracy as it is defined in *Simplicial Homotopy Theory* by Goerrs-Jardine p. 190.
+It consists of a series of maps `π₀ X → X _[0]` and `X _[n] → X _[n+1]` which
+behave formally like an extra degeneracy `σ (-1)`. It can be thought as a datum
+associated to the augmented simplicial set `X → π₀ X`.
+
+In this file, we adapt this definition to the case of augmented
+simplicial objects in any category.
+
+## Main definitions
+
+- the structure `extra_degeneracy X` for any `X : simplicial_object.augmented C`
+- `extra_degeneracy.map`: extra degeneracies are preserved by the application of any
+functor `C ⥤ D`
+- `sSet.augmented.standard_simplex.extra_degeneracy`: the standard `n`-simplex has
+an extra degeneracy
+
+TODO @joelriou:
+1) when the category `C` is preadditive and has a zero object, and
+`X : simplicial_object.augmented C` has an extra degeneracy, then the augmentation
+on the alternating face map complex of `X` is a homotopy equivalence of chain
+complexes.
+
+2) extra degeneracy for the cech nerve of a split epi. In particular the
```

Comment:
As far as I can tell, we write "Cech" in the docs and `cech` in Lean names.
```suggestion
2) extra degeneracy for the Cech nerve of a split epi. In particular the
```

Reply:
Actually we use [Čech](https://leanprover-community.github.io/mathlib_docs/topology/stone_cech.html#stone_cech). Accents are generally kept elsewhere, e.g. [deriv.lhopital_zero_at_bot](https://leanprover-community.github.io/mathlib_docs/analysis/calculus/lhopital.html#deriv.lhopital_zero_at_bot), [phragmen_lindelof.quadrant_I](https://leanprover-community.github.io/mathlib_docs/analysis/complex/phragmen_lindelof.html#phragmen_lindelof.quadrant_I)

Reply:
Good point!